pynbody/snapshot/gadgethdf.py

```diff
@@ -563,9 +563,18 @@ def __init_unit_information(self):
 
         cosmo = 'HubbleParam' in list(self._get_hdf_parameter_attrs().keys())
         if cosmo:
-            for fac in self._get_cosmo_factors(self._hdf_files[0], 'Coordinates'): dist_unit *= fac
-            for fac in self._get_cosmo_factors(self._hdf_files[0], 'Velocities'): vel_unit *= fac
-            for fac in self._get_cosmo_factors(self._hdf_files[0], 'Mass'): mass_unit *= fac
+            try:
+                for fac in self._get_cosmo_factors(self._hdf_files[0], 'Coordinates'): dist_unit *= fac
+            except:
+                dist_unit *= units.a * units.h**-1
```

Comment:
I think it would be appropriate to raise a warning here e.g.

`warnings.warn("Unable to find cosmological factors in HDF file; assuming position is %s"%dist_unit)`

or something along those lines

Reply:
I'll add all the warning messages.